Formula/s/swiftly.rb

```diff
@@ -0,0 +1,36 @@
+class Swiftly < Formula
+  desc "Swift toolchain installer and manager"
+  homepage "https://github.com/swiftlang/swiftly"
+  url "https://github.com/swiftlang/swiftly.git",
+      tag:      "1.0.0",
+      revision: "a9eecca341e6d5047c744a165bfe5bbf239987f5"
+  license "Apache-2.0"
+  head "https://github.com/swiftlang/swiftly.git", branch: "main"
+
+  # On Linux, SPM can't find zlib installed by brew.
+  depends_on :macos
+
+  depends_on macos: :ventura
+  depends_on xcode: "8.0"
+
+  uses_from_macos "swift" => :build, since: :sonoma # swift 5.10+
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "libarchive"
+  end
+
+  def install
+    args = if OS.mac?
+      ["--disable-sandbox"]
+    else
+      %w[--static-swift-stdlib -Xcc -I#{HOMEBREW_PREFIX}/include -Xlinker -L#{HOMEBREW_PREFIX}/lib]
+    end
+    system "swift", "build", *args, "--configuration", "release", "--product", "swiftly"
+    bin.install ".build/release/swiftly"
+  end
+
+  test do
+    assert_match "#{bin}/swiftly init", shell_output("#{bin}/swiftly help 2>&1", 1).chomp
```

Comment:
Can we get a test here that actually does something e.g., from the README:

```
swiftly install main-snapshot
swiftly use main-snapshot
swift --version
```

Reply:
The problem with this is that `swiftly` would install toolchains on macOS into `~/Library/Developer/Toolchains` and I assume it's not okay for a set to modify system data. In one of the upcoming versions of Swiftly, the installation location can be customized. That makes a more decent test possible.

Reply:
The tests are ran on ephemeral machines, nothing would be carried over.

Reply:
And for people running tests locally, is it okay if their environment changes?

Reply:
Very few people do that, and it would be improved in the next version.

Reply:
Fair enough. I've just improved the test.